### PR TITLE
adding fuel pump sprites and machinery to the almayer.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
Sprites have been updated by esselnek /sleepynecrons with the help of Zenith and originally made three year ago by nauticall.
Placed the fuel pump.
added some blocker behind to make it look solid.
change layer so when your behind it make sense.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: Huffie56, Zenith, esselnek/sleepynecrons, samy/nauticall, drulikar.
add: Added fuel pump as machinery on the almayer sprite will change when power is on or off.
add: Fuel pumps are animated when activated and have different state depending on how full the lifeboats are. 
imageadd: added the fuel pump icon. updated by esselnek /sleepynecrons and Zenith, they where originally made by nauticall.
maptweak: added the fuel pump to the proper locations.
/:cl:

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -351,3 +351,37 @@ Class Procs:
...
+		//if fuel pump 50
+		if(fuelpump_stage == 50)
+			icon_state = "fuelpump_50"
+		//if fuel pump 75
+		if(fuelpump_stage == 75)
+			icon_state = "fuelpump_75"
+		//if fuel pump 100
+		if(fuelpump_stage == 100)
+			icon_state = "fuelpump_100"
+//currently doesn't update properly
```
Comment (vero5123) [inline code/game/machinery/machinery.dm:387]: switch statement maybe?

File: code/controllers/subsystem/hijack.dm [hunk 2]
```diff
@@ -83,10 +83,17 @@ SUBSYSTEM_DEF(hijack)
 	/// If ARES has announced the 50% point yet for SD
 	var/ares_sd_announced = FALSE
 
+/// to fallow what stage fuel pump are in.
+var/global/fuelpump_stage = 0
```
Comment (vero5123) [inline code/controllers/subsystem/hijack.dm:87]: var/static/ is the norm I think, not var/global/

File: code/game/machinery/machinery.dm [hunk 3]
```diff
@@ -351,3 +351,37 @@ Class Procs:
 	icon_state = "autolathe"
 	density = TRUE
 	anchored = TRUE
+
+//fuel pomp
+/obj/structure/machinery/fuelpump
+	name = "\improper Fuel Pump"
+	layer = ABOVE_MOB_LAYER
+	desc = "It is a machine that pump fuel in lifeboats."
```
Comment (vero5123) [inline code/game/machinery/machinery.dm:359]: It is a machine that **pumps** fuel **around the ship**.

File: code/game/machinery/machinery.dm (hunk 1 above)
Comment (vero5123) [reply code/game/machinery/machinery.dm:387]: you know what u could do actually, if u changed the dmi iconstate from fuelpump_0 to fuelpump_1 
you could maybe just do this, should work I think.
```
if(fuelpump_stage >= 1)
      icon_state="fuelpump_[fuelpump_stage]"
```

File: code/controllers/subsystem/hijack.dm (hunk 2 above)
Comment (vero5123) [reply code/controllers/subsystem/hijack.dm:87]: Also, /global/ and GLOBAL_VAR_INIT() are not the same thing, pretty sure if you're just trying to create a "global" variable  GLOBAL_VAR_INIT(fuelpump_stage, 0) would be the right approach.

File: code/game/machinery/machinery.dm [hunk 4]
```diff
@@ -351,3 +351,37 @@ Class Procs:
...
+	icon = 'icons/obj/structures/machinery/fuelpump.dmi'
+	icon_state = "fuelpump_off"
+	density = TRUE
+	anchored = TRUE
+
+
+/obj/structure/machinery/fuelpump/update_icon()
+	if(stat & NOPOWER)
+		icon_state = "fuelpump_off"
```
Comment (kiVts) [inline code/game/machinery/machinery.dm:368]: you can probably put return here and remove the else clause completely

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by nauticall — APPROVED** [review]
nice to see my sprites getting a touchup after sitting unused in the code for 3 years *grumble grumble*

looks good tho. not your fault they gathered dust

File: code/game/machinery/machinery.dm (hunk 4 above)
Comment (Huffie56) [reply code/game/machinery/machinery.dm:371]: thanks for taking the time to help. i am going for a version without switch to all the different stage.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/blocker.dm [hunk 5]
```diff
@@ -125,3 +125,9 @@
 	icon_state = "purple_line"
 
 	visible = TRUE
+
+// for fuel pump since it's a large sprite.
+
+/obj/structure/blocker/fuelpump
+	name = "\improper Fuel Pump"
+	desc = "It is a machine that pumps fuel around the ship."
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/objects/structures/blocker.dm:133]: didn't know you could do that.

File: icons/obj/structures/machinery/fuelpump.dmi
[the remark this answers is not in the document]
Comment (Huffie56) [reply icons/obj/structures/machinery/fuelpump.dmi:1]: arg. i only tested north pumps...........

[the remark this answers is not in the document]
Comment (Drulikar) [reply icons/obj/structures/machinery/fuelpump.dmi:1]: Has nothing to do with north or south pumps; just that the pixels in the dmi are moved.

File: code/game/machinery/machinery.dm [hunk 6]
```diff
@@ -355,3 +355,19 @@ Class Procs:
...
+	icon = 'icons/obj/structures/machinery/fuelpump.dmi'
+	icon_state = "fuelpump_off"
+	density = TRUE
+	anchored = TRUE
+
+/obj/structure/machinery/fuelpump/update_icon()
+	if(stat & NOPOWER)
+		icon_state = "fuelpump_off"
+	else
+		icon_state = "fuelpump_on"
```
[the remark this answers is not in the document]
Comment (Drulikar) [reply code/game/machinery/machinery.dm]: See additional requirements above.

**Comment by Huffie56** [comment]
we could have the pump stop pumping after it reach 100... idk
animation is really good so better leave it be.

**Review by Drulikar — APPROVED** [review]
Looks to be all in order.

**Pull request merged**